node/bft/src/primary.rs

```diff
@@ -900,6 +900,11 @@ impl<N: Network> Primary<N> {
                     debug!("Skipping batch proposal {}", "(node is syncing)".dimmed());
                     continue;
                 }
+                // A best-effort attempt to skip the scheduled batch proposal if
+                // round progression already triggered one.
+                if self_.propose_lock.try_lock().is_err() {
+                    continue;
```

Comment:
Perhaps add some `debug!` message here for observing that there's an ongoing batch proposal attempt?

Reply:
It could be useful, but if my intuition is correct, such an occurrence could just be ignored (so a log would only increase verbosity); either way works for me.

Reply:
Resolved in [44f498b](https://github.com/AleoHQ/snarkOS/pull/3112/commits/44f498bc29cde5301178936115fe2835beaeebcc)